announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java

```diff
@@ -243,6 +257,29 @@ public String getEntityPrefix() {
 			log.debug("announcements.size(): {}", announcements.size());
 		}
 		
+		if (AnnouncementAction.SORT_MESSAGE_ORDER.equals(sortCurrentOrder)) {
+			try {
+				List<AnnouncementWrapper> messageList = new ArrayList<>();
+				AnnouncementChannel defaultChannel = (AnnouncementChannel) announcementService.getChannel("/announcement/channel/" + siteId + "/main");
+				for (Message msg : announcements) {
+					AnnouncementChannel curChannel = (AnnouncementChannel) announcementService.getChannel(msg.getReference().replace("msg", "channel").replaceAll("main/(.*)", "main"));
+					messageList.add(new AnnouncementWrapper((AnnouncementMessage) msg, curChannel, defaultChannel, null, null));
+				}
+				SortedIterator<AnnouncementWrapper> rvSorted = new SortedIterator<>(messageList.iterator(), new AnnouncementWrapperComparator(sortCurrentOrder, announcementSortAsc));
+				
+				List<Message> subrv = new ArrayList();
+				for (int index = 0; index < announcements.size(); index++) {
+					subrv.add(rvSorted.next());
```

Comment:
No need to loop here just use Collections.addAll

Reply:
I believe I can only add a collection to another collection.  Plus, I needed to sort messageList, thus the SortedIterator rvSorted... which is similar logic that I copied from elsewhere in AnnouncementAction.java